### PR TITLE
Add unit tests to Review and Draft

### PR DESCRIPTION
**Pull request by Pchao93**
This pull request adds RSPEC unit tests to the Review and Draft model classes. 

In addition to these unit tests, the following changes were made:
- Constantized some strings
- Changed Review uniqueness validation to use foreign keys rather than associations for testability
- Changed the Review#is_author?(user) to check for !self.user rather than !self (I don't think self should ever be falsey)
- Fixed some local flakey tests for remote_lawyers_controller and user_friend_association (let me know if these should be pulled into a separate request)

### CODE REVIEW

**Comment by Pchao93** [comment]
@CZagrobelny

File: app/controllers/regional_admin/remote_lawyers_controller.rb [hunk 1]
```diff
@@ -27,7 +27,8 @@ def destroy
 
   def user_friend_associations_params
     persisted_friend_ids = @remote_lawyer.remote_clinic_friends.map(&:id)
-    friend_ids_params = remote_lawyer_params[:friend_ids].map{ |id| id.to_i if id.present? }.compact
+    friend_ids = remote_lawyer_params[:friend_ids]
+    friend_ids_params = friend_ids ? friend_ids.map{ |id| id.to_i if id.present? }.compact : []
```
Comment (CZagrobelny) [inline app/controllers/regional_admin/remote_lawyers_controller.rb]: nice catch!

File: app/models/review.rb [hunk 2]
```diff
@@ -2,14 +2,16 @@ class Review < ApplicationRecord
...
+  validates :user_id, uniqueness: { scope: :draft_id,
+                                 message: VALIDATION_MESSAGE }
 
   scope :by_user, ->(user) {
     where(user: user)
   }
 
   def authored_by?(user)
-    !self.nil? && self.user == user
+    !self.user.nil? && self.user == user
```
Comment (CZagrobelny) [inline app/models/review.rb]: Thank you for taking a look at this.  Looking at this now, I am not 100% sure what the original intent was here.  In the two places we are calling authored_by?(user), user cannot be nil due to controller before_actions that require a current_user before the pages are rendered.  So, I think we can re-write it as:
def authored_by?(potential_author)
user == potential_author
end

Granted, this is not perfect, but I am pretty sure the nil check is extraneous in the contexts we are calling it and I needed to get rid of the local variable 'user' comparison to 'self.user'...that's just confusing 😬

File: spec/models/draft_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe Draft, type: :model do
```
Comment (CZagrobelny) [inline spec/models/draft_spec.rb]: These specs look great!

File: spec/models/user_friend_association_spec.rb [hunk 4]
```diff
@@ -7,9 +7,10 @@
   describe 'creates remote relationship' do
     let(:friend) { create(:friend) }
     let(:user) { create(:user, :remote_clinic_lawyer) }
+    let(:mailer) { double(deliver_now: true) }
 
     it 'sends email to remote lawyer' do
-      expect(FriendshipAssignmentMailer).to receive(:send_assignment).once
+      expect(FriendshipAssignmentMailer).to receive(:send_assignment).once.and_return(mailer)
```
Comment (CZagrobelny) [inline spec/models/user_friend_association_spec.rb]: Thank you for fixing this!

File: app/models/review.rb (hunk 2 above)
Comment (CZagrobelny) [reply app/models/review.rb]: @Pchao93 on second thought, I think the nil check needs to stay.  Even though we don't need it at the moment, could definitely lead to an unexpected result in the future.

**Pull request merged**